docs/development/ADRs/cartesian/frontend-indexing-absolute-k.md

```diff
@@ -0,0 +1,74 @@
+# ⚠️ Absolute K indexing
+
+In the context of porting physics parametrizations, some field accesses were at absolute indices in the K-axis. We thus decided to expand the DLS be able to port these stencils. We accept the increase in DSL surface and test it as an [experimental feature](./general-experimental-features.md).
+
+## Context
+
+Porting physics parametrizations, we found two following two access patterns
+
+```fortran
+do L = 1, LM
+  do J = 1, JM
+   do I = 1, IM
+      ...
+      Field(i, j, LM)  ! access top layer
+      Field(i, j, 1)   ! access bottom layer
+      ...
+```
+
+```fortran
+do L = 1, LM
+  do J = 1, JM
+   do I = 1, IM
+      ...
+      computed_k_index = FindK(...)  ! call a function or search all (previous) levels
+      Field(i, j, computed_k_index)  ! then, access a pre-computed absolute index (e.g. boundary layer)
+      ...
+```
+
+In some cases, porting these codes might be possible by using temporary helper fields and by breaking loop apart. In other cases (e.g. looping over previous levels) that might not be possible and the code can't be expressed as a GT4Py stencil.
```

Comment:
the previous level loops were always done with while-loops, which was not ideal but worked. 
After some thinking I could not find an example that was not describebale with the old concepts, even though doing so adds a ton of bloat and makes the code unreadable